encode.go

```diff
@@ -16,6 +17,8 @@ import (
 	"github.com/lib/pq/oid"
 )
 
+var time2400Regex = regexp.MustCompile("^(24:00(:00(\\.0+)?)?)([Z+-].*)?$")
```

Comment:
since you only use the first group, use the non-capturing group syntax for the others. i think it'd be:

`^(24:00(?::00(?:\\.0+)?)?)(?:[Z+-].*)?$`